docs/proposals/loadbalancing.md

```diff
@@ -0,0 +1,338 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/loadbalancing.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# L7 Loadbalancing and ingress
+
+## Abstract
+
+This document proposes a basic L7 loadbalancing api for kubernetes 1.1. Expect hand-waving for anything not in scope.
+
+## Motivation
+
+The proposed api changes are largely motivated by the confusion surrounding kubernetes ingress and loadbalancing witnessed on [user forums](#appendix). Users tend to reason about traffic that originates outside the cluster at the application layer. The current kubernetes api lacks L7 support alltogether, and has a very opionated networking model. This leads to an impedance mismatch high enough to drive them to port brokering.
+
+## Use Cases
+
+It goes without saying that loadbalancers play a vital role in maintaining uptime and client connectivity. Kubernetes needs an easy, HA solution for routing traffic from an external network into the cluster using application layer intelligence. Master proxy is not an acceptable solution.
+
+## Scope
+
+This proposal is limited to the kubernetes 1.1 release. L7 loadbalancing is a vast topic and we cannot commit to an api without iterating on community feedback.
+
+__What's in scope for 1.1__
+* A single application using tls termination or raw http should not need more than 1 loadbalancer to dispatch multiple url endpoints to different backend services. Eg:
+
+  ```
+  www.example.com -> |terminate ssl| 178.91.123.132 -> / foo    s1
+                                                       / bar    s2
+                                                       / foobar s3
+  ```
```

Comment:
Still reading this doc in raw form - see that you mentioned ```for 1.1``` here ... so ignore if its already addressed.